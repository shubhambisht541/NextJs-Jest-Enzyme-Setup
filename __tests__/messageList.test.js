import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import MessageList from '../components/messageList'

describe("<MessageList /> rendering ", () => {
    let wrapper,store;
    const initialState = {}
    const mockStore = configureStore()
    const props = {
        messageList:[]
    }    

    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = shallow(<MessageList store={store} {...props}/>).dive()
    })

    it("renders correctly with defualt props", () => {
        expect(wrapper).toMatchSnapshot();
    })

    it("should render one ul ", () => {
        expect(wrapper.find('ul')).toHaveLength(1);
    })

    it("ul should have className message-list ", () => {
        expect(wrapper.find('ul').hasClass('message-list')).toBe(true);
    })

})

describe('rendering on basis of messageList data', () => {
    let wrapper,store;
    const initialState = {}
    const mockStore = configureStore()
    const props = {
        messageList:[]
    }    

    beforeEach(() => {
        store = mockStore(initialState)
        wrapper = shallow(<MessageList store={store} {...props}/>).dive()
    })

    it('no <li> render in case messageList is empty', () => {
        expect(wrapper.props().messageList).toBe(undefined);
        expect(wrapper.find('li')).toHaveLength(0);
    })

    it(' <li> render in case messageList is not empty', () => {
        wrapper.setProps({ messageList: [{ id:'test', message:'message'}]})
        expect(wrapper.find('li').length).toBeGreaterThan(0);
    })

})

