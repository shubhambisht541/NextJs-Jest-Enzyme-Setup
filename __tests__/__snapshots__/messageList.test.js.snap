// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageList /> rendering  renders correctly with defualt props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageList
    dispatch={[Function]}
    messageList={Array []}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "message-list",
    },
    "ref": null,
    "rendered": Array [],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "message-list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
